docs/07_how-to/05_ingest-csv.mdx

```diff
@@ -0,0 +1,288 @@
+
+### Sink
+Once the data is in the required format, we want to sink it. For file ingress, this means updating the database. 
+
+The example sink below takes the output from `.map` (a set of DB operations) and handles the updating of the database.
+
+```kotlin
+.sink(dbSink())
+```
+
+That’s all you need to ingest the data and put it in your database.
+
+However, there are additional helpers for things such as `onCompletion`. This enables you specify what happens after the file processing is complete. 
+
+In the example below, we use this to log that we’ve completed it, and to send a notification to users:
+
+```kotlin
+.onCompletion { context ->
+    LOG.info("Completed processing of file ${context.data.name}")
+}.onCompletion(
+    notifyAllScreensOnCompletion {
+        body = "Finished processing filename ${context.data.name}"
+        header = "Finished Processing Copp Clark Exchange Holidays File"
+        severity = Severity.Information
+    }
+)
+```
+
+[comment]: <> (Waiting on error handling to add a section on it here)
+
+### What can you ingest?
+You can take data from the following sources:
+
+- PostgreSQL
+- MS SQL Server
+- Oracle Enterprise
+- Files
+  - CSV
+  - XML
+  - JSON
+
+## Three different examples
+To show how to use Data Pipeline to ingest data, we have provided an [example application](https://github.com/genesislcap/platform-howto/tree/master/howto-csv-ingress), which has three different scenarios. All of these are based on incoming data in Copp Clark holiday files:
+
+- The first example simply covers the task of parsing the incoming data (a CSV file that contains Exchange Settlement details) and then updating a table in the application's database (the EXCHANGE_HOLIDAY table).
+
+- The second example looks at two incoming files - Trading and Settlement - and applies more sophisticated analysis and logic. The database is only updated when changes are detected, including deleting dates that are no longer designated holidays. In this example, the logic is defined in a separate transformer file, which is called from within the pipeline itself.
+
+- The third example is an extension of the second. It adds maintenance of EXCHANGE data, which is contained in every row of the inbound files. 
+
+The application also includes guidance for:
+
+- Sourcing from another location
+    - Remove SFTP serving
+    - AWS S3 Bucket
+
+Throughout, the code in the application includes detailed comments explaining the steps.
+
+#### Finding the examples
+The examples are within one complete example application, which includes a front end so that you can run and see the data. You can [clone the repo](https://github.com/genesislcap/platform-howto/tree/master/howto-csv-ingress) to see the code - which includes comments at the key points to highlight what is being specified - and run the application.
```

Comment:
Need to be careful of this and ANY reference in ANY file to github.com/genesislcap/ as this is internal to us and will be a dead link to all clients . We need to know the resulting URLS and scan all these files and update them, if we do it right we should only need to change /genesislcap/ to wherever we decided to publish (communitysuccess?)

Reply:
to be handled by
https://genesisglobal.atlassian.net/browse/PTL-1417